Updating to latest release of raylib

damntourists opened the pull request:
This PR adds all of the latest updates from https://github.com/raysan5/raylib which I've also confirmed as a fix for https://github.com/gen2brain/raylib-go/issues/521.

damntourists committed:
Updating to latest release of raylib.

damntourists committed:
Merge branch 'gen2brain:master' into issue-521-update_c_sources

damntourists committed:
remove external glfw deps and rglfw.c file which are not present in gen2brain/raylib-go.

damntourists committed:
Added SDL2 and SDL3 separation for build and test. Added macos and windows to test-sdl CI job. Added test-sdl3 CI job.

damntourists committed:
Update test-sdl to call sdl2 build tag instead of sdl.

damntourists committed:
Update sdl build tag name to sdl2 to avoid confusion.

damntourists committed:
Add -DSDL_UNIX_CONSOLE_BUILD=ON to test-sdl3 github action job.

damntourists committed:
Fix test-sdl3 for windows. should be all lowercase: mingw-w64-x86_64-sdl3

damntourists committed:
Fix test-sdl3 for windows. Need to include PATH variable.

damntourists committed:
Fix test-sdl3 for windows. Path variable empty. Try another method.

damntourists committed:
Reverting build.yml and sdl2 build tag reverted to original name of sdl.

damntourists committed:
Add shim for sdl2 builds.

damntourists committed:
Add sdl_shim.h for sdl2 builds.

damntourists commented:
I added a sdl_shim.h which enables compiling with sdl2. Previously was failing with 
```
In file included from rcore.c:545:
platforms/rcore_desktop_sdl.c: In function ‘PollInputEvents’:
platforms/rcore_desktop_sdl.c:1753:31: error: ‘SDL_Event’ has no member named ‘gbutton’; did you mean ‘button’?
 1753 |                 switch (event.gbutton.button)
      |                               ^~~~~~~
      |                               button
platforms/rcore_desktop_sdl.c:1781:60: error: ‘SDL_Event’ has no member named ‘gbutton’; did you mean ‘button’?
 1781 |                         if (platform.gamepadId[i] == event.gbutton.which)
      |                                                            ^~~~~~~
      |                                                            button
platforms/rcore_desktop_sdl.c:1794:31: error: ‘SDL_Event’ has no member named ‘gbutton’; did you mean ‘button’?
 1794 |                 switch (event.gbutton.button)
      |                               ^~~~~~~
      |                               button
platforms/rcore_desktop_sdl.c:1822:60: error: ‘SDL_Event’ has no member named ‘gbutton’; did you mean ‘button’?
 1822 |                         if (platform.gamepadId[i] == event.gbutton.which)
      |                                                            ^~~~~~~
      |                                                            button
platforms/rcore_desktop_sdl.c: In function ‘InitPlatform’:
platforms/rcore_desktop_sdl.c:2058:33: warning: initialization of ‘SDL_JoystickID *’ {aka ‘int *’} from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
 2058 |     SDL_JoystickID *joysticks = SDL_GetJoysticks(&numJoysticks); // array of joystick IDs, they do not start from 0
      |                                 ^~~~~~~~~~~~~~~~
      ```

gen2brain commented:
It has always been able to compile for SDL2 and, recently, for SDL3. But rcore_desktop_sdl.c changed, that happens on every update, everything changes. See the history, etc. Does raylib use that shim.h? If not, then the issue is elsewhere, and we should not hack it or add unnecessary files.

damntourists commented:
It seems the issue was possibly introduced upstream in https://github.com/raysan5/raylib/pull/5176/files since this is using SDL3 specific code. In order to get SDL2 to work with it, the shim was created.

damntourists committed:
Revert README.md changes.

damntourists committed:
Revert README.md changes.

damntourists committed:
Merge remote-tracking branch 'origin/issue-521-update_c_sources' into issue-521-update_c_sources

damntourists committed:
Adding fixes from https://github.com/raysan5/raylib/issues/5403

damntourists committed:
Reverting build.yml

damntourists committed:
Removing previously added files for sdl3 workaround. Reverting cgo_platformHere_sdl.go files to originals.

damntourists committed:
Fix for cgo_linux_drm.go compilation issues regarding uint type.

damntourists committed:
Merge branch 'gen2brain:master' into issue-521-update_c_sources

damntourists commented:
@gen2brain, I've made some updates to this PR. The issue that I was attempting to fix with the shim has been corrected upstream in raylib which I've updated in raylib-go. All jobs in the package's workflow passed in my latest commit @ https://github.com/damntourists/raylib-go/actions/runs/20114236373. Could you take a look and let me know if there's anything I've missed?

gen2brain commented:
Nice. Please check https://github.com/gen2brain/raylib-go/pull/472; a few things must always be changed manually. Also, I think I saw in some raylib commits that something changed for the Android main entry point, but I'm not sure; I do not have time to check now.

damntourists committed:
Add undefine ALIGN if defined for jar_mx.h and updated android_main entrypoint in rcore_android.c per 5ed790526c258c796a6bbba5cab55b8f4b600294.

damntourists committed:
Add !noaudio build tag to raudio.c

damntourists commented:
Okay, I've gone through and added the ALIGN def/undef block to jar_mx.h, the !noaudio build tag to raudio.c and updated the entry point for rcore_android.c. Whenever you get a moment, could you check on your end if there's anything I've missed? The unittests are still passing on my fork.

gen2brain commented:
Nice, thanks for working on this. I am merging it; it looks good. It is always good to do this, and it is much easier later on a new release. Users still get the latest tagged release; they do not have to use main/master.
